### PR TITLE
PBKDF2: Support ArrayBuffer for "password" field

### PR DESCRIPTION
**Pull request by majorhayes**
## Changes
* Change native code for PBKDF2 to always take base64 strings for "password"
* For PBKDF2, rename the last parameter from "hash" to "algorithm" to avoid confusion with the function name of "hash"
* Update iOS deployment target to 10.0 because some of the iOS crypto functions used in the lib require 10.0
* Update README

## Refactors
* Android: Move duplicate "bytesToHex" function into a Util class 
* Fix reference to Base64 to use the android library instead of java
* Remove unused imports (Android Studio -> Optimize Imports)

### CODE REVIEW

**Comment by majorhayes** [comment]
Could I get a review on this?

**Comment by ghbutton** [comment]
Yea will do

**Comment by ghbutton** [comment]
Could you update the README with an example with the new code?

File: android/src/main/java/com/pedrouid/crypto/RCTAes.java [hunk 1]
```diff
@@ -96,24 +96,13 @@ public void randomKey(Integer length, Promise promise) {
             byte[] key = new byte[length];
             SecureRandom rand = new SecureRandom();
             rand.nextBytes(key);
-            String keyHex = bytesToHex(key);
+            String keyHex = Util.bytesToHex(key);
```
Comment (ghbutton) [inline android/src/main/java/com/pedrouid/crypto/RCTAes.java:99]: 👍

**Comment by majorhayes** [comment]
Updated the README. Sorry, I thought I had done that already.

**Comment by majorhayes** [comment]
I included the change from #22 in here also. His line numbers didn't match up with my master so I wanted to avoid a conflict after merging.

**Comment by cjthompson** [comment]
Is there anything else you'd like me to do to get this PR ready to merge?

**Comment by ghbutton** [comment]
No, I will test and merge it soon

**Pull request merged**